doc/rfd/rfd/0009/README.md

```diff
@@ -0,0 +1,176 @@
+---
+authors: Victor Cuadrado <vcuadradojuan@suse.com>, Ettore Di Giacinto <edigiacinto@suse.de>
+state: discussion
+discussion: https://github.com/cloudfoundry-incubator/kubecf/pull/1495
+---
+
+# RFD 9: Specification for deployment and automation scripts
+
+## Context
+
+
+Dealing with a handful of loosely dependent Helm charts and building a Helm
+distribution from them without creating an all-or-nothing "super-chart" is
+difficult. One needs to make sure that deployment and automation scripts are
+interoperable between charts, repos, repos' histories, and that they can be
+reused and extended downstream as needed.
+
+At the same time one needs to create, reuse, and extend test environments across
+charts, and between upstream and downstream.
+
+The following spec is an interface specification for sharing deployment and
+automation scripts of Helm charts and utility applications.
+
+
+## Specification
+
+### Version `0.1`.
+
+
+The interface consists in the following structure of folders and executables,
+together with their expected use:
+
+```
+├── clean/*
+├── chart/*
+├── configure/*
+│   └── patches/*
+├── install/*
+├── wait/*
+├── include/*
+│
+├── one-offs/*
+│
+├── include/*
+├── example-values/*
+│
+└── spec_ver
+```
+
+This structure as a whole is encouraged to be published together with the Helm
+chart. This can be for example as artifacts of a chart release, or present in
+the chart repository, but we recommend that this files are bundled inside of the
+chart: this simplifies consumption downstream.
+
+While targets are allowed to take arguments and options on execution besides the
+expected default behaviour defined here, this is discouraged.
+
+Each target is expected to be a folder (E.g: `install/`) containing executable
+files starting with digits (E.g: `10_do-foo.sh`, `20_do-bar.rb`).
+
+Upon target call, all the executable files starting with digits will get
+executed in alphanumeric order.
+It is allowed for the folder to contain other files/folders besides those
+starting with digits. Those would not be automatically executed upon target
+call (e.g: readme files, local include folders, etc).
+  
+Deployment and automation scripts will expose the following targets:
+
+- `chart`: Obtains the latest public release of the chart at hand, and puts it
+  already decompressed in `$pwd/$our-chart/chart/`. `$our-chart` being the name
+  of the chart at hand (e.g: kubecf).
+
+- `configure`: Outcome is a valid `$pwd/$our-chart/values.yaml` file for the
```

Comment:
Why mangle yaml files? The `helm` CLI supports specifying multiple values yaml files.

Reply:
That's correct, and a better approach. I think a good approach would be for `configure` target to provide a `$pwd/values/` folder with those, that the `install` and `upgrade` targets would consume then.

Reply:
I'm still thinking on how to tackle this properly. Merging subsets of yaml files has the benefit that one can order then alphanumerically, and provide a deterministic way to do the merge.
I wonder about yaml files whose objects overlap. Do any charts have any of those? What happens when merging them? Does it matter?

Reply:
Moved to a folder for all config values in https://github.com/cloudfoundry-incubator/kubecf/pull/1495/commits/3e5206b7c0cadbb2d136701b481830e789b34fe8